packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -396,6 +396,14 @@ export interface DataGridPropsWithDefaultValues<R extends GridValidRowModel = an
    * @default false
    */
   unstable_rowSpanning: boolean;
+  /**
+   * If `true`, the Data Grid doesn't disable column virtualization when `getRowHeight` is set to `() => 'auto'`.
+   * By default, column virtualization is disabled when dynamic row height is enabled to measure the row height correctly.
+   * For datasets with a large number of columns, this can cause performance issues.
+   * The downside of enabling this prop is that the row height will be estimated based the cells that are currently rendered, which can cause row height change when scrolling horizontally.
```

Comment:
Love that, giving the control of the UX tradeoff in the hands of designers.

Reply:
We will see how this goes. It looks like AG Grid doesn't support this option https://www.ag-grid.com/javascript-data-grid/row-height/#auto-row-height. Still, experiencing this https://codesandbox.io/p/sandbox/focused-rain-zxyt6p?file=%2Fdemo.js%3A21%2C35, I almost wonder if this shouldn't be the default. It's easier to search for

"layout shift with auto height" than "grid slow performance" 😄. Anyway, we have this in the docs. We will see.

Reply:
> It looks like AG Grid doesn't support this option [ag-grid.com/javascript-data-grid/row-height#auto-row-height](https://www.ag-grid.com/javascript-data-grid/row-height/#auto-row-height).

I think they don't need it – they took an interesting approach of per-column row auto height.
So they can exclude specific columns from virtualization (i.e. always render them) if they have auto height enabled.
It's smart.